tests/src/test/scala/common/Wsk.scala

```diff
@@ -1036,13 +1036,28 @@ trait RunWskCmd extends BaseRunWsk {
     val args = baseCommand
     if (verbose) args += "--verbose"
     if (showCmd) println(args.mkString(" ") + " " + params.mkString(" "))
-    val rr = TestUtils.runCmd(
-      DONTCARE_EXIT,
-      workingDir,
-      TestUtils.logger,
-      sys.env ++ env,
-      stdinFile.getOrElse(null),
-      args ++ params: _*)
+    val rr =
+      retry(
+        {
+          val rr = TestUtils.runCmd(
+            DONTCARE_EXIT,
+            workingDir,
+            TestUtils.logger,
+            sys.env ++ env,
+            stdinFile.getOrElse(null),
+            args ++ params: _*)
+
+          if (expectedExitCode != NETWORK_ERROR_EXIT) {
+            withClue(hideStr(reportFailure(args ++ params, expectedExitCode, rr).toString(), hideFromOutput)) {
```

Comment:
is the `withClue` line still needed now that the `retry` is displaying the failure message?

Reply:
Yes, `withClue` will print the stack trace if the final retry fails.

Reply:
will each retry will have a `withClue` stack trace message, followed by a final `withClue` message (line 1062) if all retries fail?

Reply:
Just a final retry that fails will have the stack trace.

Reply:
This change in my opinion is wrong and should be backed out.

The retry method used here requires the function under retry to throw an exception to be retried, and use of `retry` here makes it impossible to exit the retry loop subject to the expected exit code unless that code is `NETWORK_ERROR_EXIT`.

I suggest backing this out and then fixing it properly, with a test for the `cli` command itself.

FYI @houshengbo

Reply:
I'm addressed this in #3776.